nix/process-compose-flake-lib.nix

```diff
@@ -0,0 +1,99 @@
+{ lib }:
+let
+  inherit (lib) types;
+in
+{
+  mkProcessComposeArgumentsOption = {}:
+    lib.mkOption {
+      type = types.submodule
+        ({ config, lib, ... }:
+          let inherit (lib) types mkOption;
+          in
+          {
+            options = {
+              config = mkOption {
+                type = types.listOf types.str;
+                default = [ ];
+              };
+              detached = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              disable-dotenv = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              env = mkOption {
+                type = types.listOf types.str;
+                default = [ ];
+              };
+              hide-disabled = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              keep-project = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              namespace = mkOption {
+                type = types.listOf types.str;
+                default = [ ];
+              };
+              no-deps = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              ref-rate = mkOption {
+                type = types.str;
+                default = "";
+              };
+              reverse = mkOption {
+                type = types.bool;
+                default = false;
+              };
+              sort = mkOption {
+                type = types.str;
+                default = "";
```

Comment:
Instead of using empty strings (which has ambiguous meaning, since you are forced to look at the implementation to understand the semantics), we should use `types.nullOr types.str` as type in options like this.

Reply:
I've used nullable strings in the past as well, but don't do that nowadays, except if null and "" has different meaning. In this case, it doesn't. The string being nullable only adds the burden of needing to check if it's null in addition to needing to check if it's empty. Same goes for arrays, i always default to [] instead of null for the same reasons.

But in the end it's a matter of taste and I'll follow your command.

Reply:
> But in the end it's a matter of taste and I'll follow your command.

Ok.